File: MessageApp/app/src/main/java/com/example/messageapp/MainActivity.java
package com.example.messageapp;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import android.Manifest;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.telephony.SmsManager;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
     EditText mno,message;
     Button send;
     String mobno,mess;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mno = findViewById(R.id.mno);
        message = findViewById(R.id.msg);
        send = findViewById(R.id.send);
        String mobno = mno.getText().toString();
        String mess = message.getText().toString();
        send.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String mobno = mno.getText().toString();
                String mess = message.getText().toString();
                sendSMS(mobno,mess);
            }
        });
    }
    public void sendSMS(String userNumber,String userMessage)
    {
        if(ContextCompat.checkSelfPermission(this, Manifest.permission.SEND_SMS)!= PackageManager.PERMISSION_GRANTED){
            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.SEND_SMS},1);
        }
        else
        {
            SmsManager smsManager = SmsManager.getDefault();
            smsManager.sendTextMessage(userNumber,null,userMessage,null,null);
        }
    }
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,@NonNull int[] grantResults){
        super.onRequestPermissionsResult(requestCode,permissions,grantResults);
        if (requestCode == 1 && grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
            SmsManager smsManager = SmsManager.getDefault();
            smsManager.sendTextMessage(mobno,null,mess,null,null);
        }
    }

}

File: ImageCaptureApplication/app/src/main/java/com/rajalakshmi/imagecaptureapplication/MainActivity.kt
package com.rajalakshmi.imagecaptureapplication

import android.Manifest
import android.R.attr
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.core.app.ActivityCompat.startActivityForResult

import androidx.core.app.ActivityCompat.requestPermissions

import android.content.pm.PackageManager
import android.os.Build
import android.provider.MediaStore
import android.widget.ImageView
import androidx.annotation.RequiresApi
import android.graphics.Bitmap

import android.app.Activity
import android.util.Log

import androidx.annotation.NonNull
import android.R.attr.data
import androidx.activity.result.contract.ActivityResultContracts
import androidx.core.app.ActivityCompat.requestPermissions
import android.R.attr.data








class MainActivity : AppCompatActivity() {

    private val CAMERA_REQUEST = 1888
    private var imageView: ImageView? = null
    private val MY_CAMERA_PERMISSION_CODE = 100

    @RequiresApi(Build.VERSION_CODES.M)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imageView = findViewById(R.id.imageView)

        val ibtn_profile = findViewById<Button>(R.id.btn_select_image)
        ibtn_profile.setOnClickListener{
            if (checkSelfPermission(Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
                    val permission = arrayOf(Manifest.permission.CAMERA)
                    requestPermissions(permission, MY_CAMERA_PERMISSION_CODE)


            } else {
                val cameraIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
                startActivityForResult(cameraIntent, CAMERA_REQUEST)
            }

        }
    }


    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if(requestCode==1888 && resultCode== Activity.RESULT_OK && data!=null){
            Log.i("info", "Photo Selected")
            var bitmap : Bitmap = data.extras!!.get("data") as Bitmap
            Log.i("SRI1711",bitmap.toString())
            imageView!!.setImageBitmap(bitmap)



        }
    }


}